### ISSUE TITLE
Script and integration tests to reproduce selected results from Transformers4Rec paper with Merlin Models API

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/models#806]
### Description
For Transformers4Rec, we have created a training/eval [script](https://github.com/NVIDIA-Merlin/Transformers4Rec/tree/main/examples/t4rec_paper_experiments) for reproducing the paper experiments, that takes a set of hparams as command line arguments and a preprocessed dataset.

This task is about creating the training/eval script based on the original Transformers4Rec script using the Merlin Models API, which will:
- Ensure that our TF implementation is correct and that it matches the results we had using Transformers4Rec (PyTorch)
- Work as an advanced example on how to set the available hparams for session-based recommendation
- Work as integration tests to ensure the stability of the API for setting the advanced hparams and also allow tracking in our CI  accuracy or performance regressions (like we have for [retrieval integration tests](https://github.com/NVIDIA-Merlin/models/issues/482)).

**Tasks**

- [ ] Implement training/eval script to reproduce GPT-2 (CLM), TransformerXL(CLM)  and XLNet (MLM) (with and without features)  results from Transformer4Rec paper for REES46 dataset
- [ ] Implement unit/integration tests to check in the CI for API breaks, and accuracy/perf regressions

**Comment by viswa-nvidia** [comment]
@gabrielspmoreira , please move this to another RMP ticket
